spring-integration-gcp/src/main/java/org/springframework/integration/gcp/outbound/PubSubMessageHandler.java

```diff
@@ -17,50 +17,29 @@
 
 package org.springframework.integration.gcp.outbound;
 
-import java.io.IOException;
+import com.google.auth.oauth2.GoogleCredentials;
 
-import com.google.cloud.pubsub.spi.v1.Publisher;
-import com.google.pubsub.v1.PubsubMessage;
-import com.google.pubsub.v1.TopicName;
-
-import org.springframework.cloud.gcp.pubsub.converters.SimpleMessageConverter;
+import org.springframework.cloud.gcp.pubsub.core.PubSubTemplate;
 import org.springframework.integration.handler.AbstractMessageHandler;
 import org.springframework.messaging.Message;
-import org.springframework.messaging.converter.MessageConversionException;
-import org.springframework.messaging.converter.MessageConverter;
 
 /**
- * Converts from internal Spring message to GCP Pub/Sub message and publishes the Pub/Sub
- * message to a Pub/Sub topic.
+ * Sends messages to Google Cloud Pub/Sub by delegating to {@link PubSubTemplate}.
  *
  * @author João André Martins
  */
 public class PubSubMessageHandler extends AbstractMessageHandler {
 
-	private Publisher publisher;
-	private MessageConverter messageConverter;
-
-	public PubSubMessageHandler(String projectId, String topicName) throws IOException {
-		this(projectId, topicName, new SimpleMessageConverter());
-	}
+	private PubSubTemplate pubsubTemplate;
+	private String topic;
 
-	public PubSubMessageHandler(String projectId, String topicName,
-			MessageConverter messageConverter) throws IOException {
-		publisher = Publisher.defaultBuilder(TopicName.create(projectId, topicName))
-				.build();
-		this.messageConverter = messageConverter;
+	public PubSubMessageHandler(String projectId, String topic, GoogleCredentials credentials) {
+		pubsubTemplate = new PubSubTemplate(credentials, projectId);
+		this.topic = topic;
 	}
 
 	@Override
 	protected void handleMessageInternal(Message<?> message) throws Exception {
-		Object pubsubMessageObject = messageConverter.fromMessage(message,
-				PubsubMessage.class);
-
-		if (!(pubsubMessageObject instanceof PubsubMessage)) {
-			throw new MessageConversionException("The specified converter must produce"
-					+ "PubsubMessages to send to Google Cloud Pub/Sub.");
-		}
-
-		publisher.publish((PubsubMessage) pubsubMessageObject);
+		pubsubTemplate.send(topic, message);
```

Comment:
Just for reference, the checkstyle file isn't enforcing `this` on property references, I wonder if it supports this.